NtupleProducer/python/scouting/runW3PiNtuples_cfg.py

```diff
@@ -0,0 +1,173 @@
+process.load("SimGeneral.HepPDTESSource.pythiapdt_cfi")
+process.load("FWCore.MessageLogger.MessageLogger_cfi")
+process.options = cms.untracked.PSet(
+            wantSummary = cms.untracked.bool(True),
+)
+process.maxEvents = cms.untracked.PSet( input = cms.untracked.int32(10))
+process.MessageLogger.cerr.FwkReport.reportEvery = 10
+
+process.source = cms.Source("PoolSource",
+    fileNames = cms.untracked.vstring('file:/tmp/gpetrucc/005E74D6-B50E-674E-89E6-EAA9A617B476.root')
+)
+
+process.load('Configuration.Geometry.GeometryExtended2026D88Reco_cff')
+process.load('Configuration.Geometry.GeometryExtended2026D88_cff')
+process.load('Configuration.StandardSequences.MagneticField_cff')
+process.load('Configuration.StandardSequences.SimL1Emulator_cff')
+process.load('SimCalorimetry.HcalTrigPrimProducers.hcaltpdigi_cff') # needed to read HCal TPs
+process.load('SimGeneral.MixingModule.mixNoPU_cfi')
+process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_cff')
+
+from Configuration.AlCa.GlobalTag import GlobalTag
+process.GlobalTag = GlobalTag(process.GlobalTag, '125X_mcRun4_realistic_v2', '')
+
+process.extraPFStuff = cms.Task(
+        process.l1tSAMuonsGmt,
+        process.l1tGTTInputProducer,
+        process.l1tVertexFinderEmulator,
+        process.L1TLayer1TaskInputsTask,
+        process.L1TLayer1Task,
+        process.L1TPFJetsEmulationTask
+)
+
+process.l1tLayer2Deregionizer.nPuppiFinalBuffer = 200
+
+process.puppiTable = cms.EDProducer("SimpleCandidateFlatTableProducer",
+                src = cms.InputTag("l1tLayer2Deregionizer:Puppi"),
+                cut = cms.string(""),
+                name = cms.string("L1Puppi"),
+                doc = cms.string("L1Puppi candidates"),
+                singleton = cms.bool(False), # the number of entries is variable
+                extension = cms.bool(False), # this is the main table
+                variables = cms.PSet(
+                    pt   = Var("pt",  float,precision=8),
+                    phi  = Var("phi", float,precision=8),
+                    eta  = Var("eta", float,precision=8),
+                    mass = Var("mass", float,precision=8),
+                    vz   = Var("vz",  float,precision=8),
+                    charge = Var("charge", int, doc="charge"),
+                    pdgId  = Var("pdgId", int, doc="PDG id"),
+                )
+            )
+
+process.p = cms.Path(
+    process.puppiTable
+)
+process.p.associate(process.extraPFStuff)
+
+def goSignal():
+    ## Add to the configuration:
+    ##   * the generator-level W boson and pions
+    ##   * for each Puppi candidate, information on whether it's matched to a gen pion
+    ##   * the table of PF candidates, so that one can 
+    process.genW = cms.EDFilter("GenParticleSelector",
+            src = cms.InputTag("genParticles"),
+            cut = cms.string("abs(pdgId) = 24 && numberOfDaughters = 3 && abs(daughter(0).pdgId) = 211 && abs(daughter(1).pdgId) = 211"),
```

Comment:
Shouldn't this cut consider the 3<sup>rd</sup> daughter as well?
```suggestion
            cut = cms.string("abs(pdgId) = 24 && numberOfDaughters = 3 && abs(daughter(0).pdgId) = 211 && abs(daughter(1).pdgId) = 211 && abs(daughter(2).pdgId) = 211"),
```

Reply:
well, in these MC events the W's either decay to 3 pions or have some "internal" decay like W -> W + gamma that pythia uses to denote radiation, so checking 2 was already sufficient, but indeed checking the third doesn't hurt and makes it more readable